firefly/app.py

```diff
@@ -28,14 +29,21 @@ def __init__(self, function, **kwargs):
 
     def __call__(self, request):
         kwargs = self.get_inputs(request)
-        result = self.function(**kwargs)
-        return self.make_response(result)
+        try:
+            validate_args(self.function, kwargs)
+            result = self.function(**kwargs)
+            return self.make_response(result)
+        except ValidationError as err:
```

Comment:
the function  should be called only after the validation is done. Move the function call after the except block.